kiwi_boxed_plugin/tasks/system_boxbuild.py

```diff
@@ -181,15 +182,26 @@ def _validate_kiwi_build_command(self):
         )
         if '--' in kiwi_build_command:
             kiwi_build_command.remove('--')
+        # validate build command through docopt from the original
+        # kiwi.tasks.system_build docopt information
         log.info(
             'Validating kiwi_build_command_args:{0}    {1}'.format(
                 os.linesep, kiwi_build_command
             )
         )
-        docopt(
+        validated_build_command = docopt(
             doc=kiwi.tasks.system_build.__doc__,
             argv=kiwi_build_command
         )
+        # rebuild kiwi build command from validated docopt parser result
+        kiwi_build_command = [
+            'system', 'build'
+        ]
+        for option, value in validated_build_command.items():
+            if option.startswith('-') and value:
+                kiwi_build_command.append(option)
+                if isinstance(value, str):
```

Comment:
>I believe this is not sufficient. What for --add-package or other flags that can be passed multiple times? And what for boolean flags?

bool flags are covered by ```if isinstance(value, str):```

I have not tested on arguments that can be passed multiple times... will check that

Reply:
> bool flags are covered by `if isinstance(value, str):`

Oh I see, you prepend the option and later check the value type. I overlooked that, in any case I bet that for arguments that can be passed multiple times they will be ignored, since docopt returns a list even if they are passed only once so the value won't ever be of type str.

Reply:
yes you are right, thanks I'll adapt the code according to your suggestion

Reply:
done, how about it now ?